dev/bots/test.dart

```diff
@@ -851,6 +853,75 @@ Future<void> _runWebReleaseTest(String target, {
   }
 }
 
+// Run a web service worker test. The expectations are currently stored here
+// instead of in the application.
+Future<void> _runWebServiceWorkerTest(String target, {
+  List<String> additionalArguments = const<String>[],
+}) async {
+  final String testAppDirectory = path.join(flutterRoot, 'dev', 'integration_tests', 'web');
+  final String appBuildDirectory = path.join(testAppDirectory, 'build', 'web');
+
+  // Build the app.
+  await runCommand(
+    flutter,
+    <String>[ 'clean' ],
+    workingDirectory: testAppDirectory,
+  );
+  await runCommand(
+    flutter,
+    <String>[
+      'build',
+      'web',
+      '--release',
+      ...additionalArguments,
+      '-t',
+      target,
+    ],
+    workingDirectory: testAppDirectory,
+    environment: <String, String>{
+      'FLUTTER_WEB': 'true',
+    },
+  );
+  final List<Uri> requests = <Uri>[];
+  final List<Map<String, String>> headers = <Map<String, String>>[];
+  await runRecordingServer(
+    appUrl: 'http://localhost:8080/',
+    appDirectory: appBuildDirectory,
+    requests: requests,
+    headers: headers,
+    browserDebugPort: null,
+  );
+
+  final List<String> requestedPaths = requests.map((Uri uri) => uri.toString()).toList();
+  final List<String> expectedPaths = <String>[
+    // Initial page load
+    '',
+    'main.dart.js',
+    'assets/FontManifest.json',
+    'flutter_service_worker.js',
+    'manifest.json',
+    'favicon.ico',
+    // Service worker install.
+    'main.dart.js?id=18f7de7f7ebd8ff11a8e835cd99d45bc',
```

Comment:
Note: I need a better solution here since this will change every time dart rolls